### DISCUSSION TITLE
parameter transformation in pmcmc

### DISCUSSION THREAD

**Discussion by Fuhan-Yang** [kingaa/pomp#191] [Q&A] [answered]
I have a simple question about the parameter transformation in `pmcmc()`. How to correctly specify the parameter transformation in `pmcmc()`?

My pomp model is a seasonal SIRS model. I have four parameters, parameters `beta`, `seas`, `rho` are between 0 and 1, and `pi` is between 100 to 800. I tried to transform the proposal distribution, but it won't transform back in `pmcmc` given the transformation is specified in `partrans` argument. Here is my code: 

```
### pmcmc trial ###

## proposal distribution #
S <- diag(1, nrow = 4, ncol = 4, names = T)

rownames(S) <- c("beta","seas","rho","pi")
colnames(S) <- c("beta","seas","rho","pi")

mvn_rw(rw.var = S) -> foo

### pmcmc ###
efit_mcmc <- sirs_sto_annual_beta |>
  pomp(
    params = c(beta = 0.22, seas = 0.001, rho = 0.006, 
      pi = 100, mu_IR = 0.2, N = 1e7)
  ) |>
  pmcmc(
    paramnames = c("beta","seas","rho","pi","gamma","N","T_beta","T_rho",""T_seas,"T_pi"),
    partrans = parameter_trans(
      logit = c("beta","seas","rho"),
      toEst=Csnippet("T_pi = logit(pi/800-0.125);"),
      fromEst=Csnippet("pi = 800*(0.125+expit(T_pi));")
    ),
    Nmcmc = 10,
    proposal = foo,
    Np = 100, 
    dprior = NULL,
    verbose = FALSE
  )
```
Thank you so much!

**Comment by kingaa** [comment, marked as answer]
`pmcmc` ignores parameter transformations.  The reason for this is that, if one changes coordinates on the parameter space, one must also transform the prior distribution, which is not as simple as applying a point transformation.  Note that, according to the [manual page](https://kingaa.github.io/manuals/pomp/html/pmcmc.html), `pmcmc` does not take a `partrans` argument.

This also shows the way to accommodate constrained parameter spaces: simply let your prior reflect the constraints on the parameters.  That is, make sure that the prior is zero for impermissible parameters.

**Reply by Fuhan-Yang** [reply]
Thanks for the fast reply. After the constrained prior is added, the chain is still not running because the binomial likelihood fails to evaluate when the negative parameters are proposed. Now the question is, can I constrain the proposal to be between 0 and 1 while allowing enough exploration of parameter space?

**Reply by kingaa** [reply]
You can supply a custom proposal.  At present, **pomp** supports only symmetric proposals.

**Reply by Fuhan-Yang** [reply]
I logited my parameters in the pomp model to remove the constraints on the proposal, because I will try adaptive tuning, which needs normal proposal too. Hope this won't bring issues. Thanks for your help! I edited my code to make the question concise.
